### PR TITLE
Windows 10 Calculator (follow-up to PR #9429): results announcement in compact overlay and unit converter

### LINKED ISSUE #11880: Windows 10 Calculator: repetition in unit converter
Hi,

Follow-up to PR #9429 

### Steps to reproduce:
In Windows 10 Calculator:

1. With Calculator set to full mode (Alt+Down arrow), press Alt+H to open the menu.
2. Choose any of the unit converters (volume, distance, etc.).
3. While focused on either "convert from" or 'converts into" fields, type some numbers (do not press Enter as unit conversion is realtime).
4. Perform some more unit conversions, especially trying out the same units and same values.

### Actual behavior:
NVDA announces conversion values up to two times. In these instances, NVDA will first announce the source and target values together (due to UIA notification event), followed by the value under the focus (convert from/converts into).

### Expected behavior:
Result is announced only once via UIA notification event.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-21429,874230a7

#### Windows version:
Windows 10 Version 20H2 (build 19042.662 beta)

#### Name and version of other software in use when reproducing the issue:
Windows Calculator 10.2010.0.0

#### Other information about your system:
Development workstation for Windows 10 App Essentials add-on

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Same in 2020.3

#### If addons are disabled, is your problem still occurring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Not applicable

### Additional context
This is caused by cache checks in name change event, specifically the results cache is not cleared when a user presses Escape, Enter, or Numpad Enter (results script).

### Solutions
The obvious solution is clearing results cache. A more subtle solution is adding unit conversion source and target Automation Id's to a list of Id's to be suppressed (when announcing results). The upcoming PR (also see the next issue) will combine these.

Thanks.

### LINKED ISSUE #11881: Windows 10 Calculator: results are not announced in compact overlay mode
Hi,

Follow-up to PR #9429 

### Steps to reproduce:
In Windows 10 Calculator:

1. Switch to standard calculator mode (Alt+number row 1).
2. Press Alt+Up arrow to switch to always on top (compact overlay) mode.
3. Perform some calculations.

### Actual behavior:
Results are not announced.

### Expected behavior:
Results are announced.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-21429,874230a7

#### Windows version:
Windows 10 Version 20H2 (build 19042.662 beta)

#### Name and version of other software in use when reproducing the issue:
Calculator 10.2010.0.0

#### Other information about your system:
N/A

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Same behavior in 2020.3.

#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Not applicable

### Additional context and solution
Caused by a missing Automation Id that should be included in result announcement script (UIA Automation Id is different for always on top result element). The same PR for #11880 will be applicable to this issue as well.

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Follow-up to #9429 
Fixes #11880 
Fixes #11881 

### Summary of the issue:
Improve result announcements in compact overlay mode and unit converters. Specifically:

* NVDA will now announce calculator results in compact overlay (alwasy on top) mode.
* NVDA will not announce convert from/converts into fields multiple times in unit converter.

### Description of how this pull request fixes the issue:
For compact overlay: add a UIA Automation Id used in alwasy on top mode to represent calculator results, resulting in results being announced when Enter, Numpad Enter, or Escape is pressed.

Unit converter: NVDA will no longer announce convert from/converts into fields multiple times, resolved by adding "Value1" and "Value2" Automation Id strings for unit conversion fields to a list of no result announcement list.

### Testing performed:
Tested via Windows 10 App Essentials and from source code with Calculator version 10.2010 and later running on Windows 10 Version 20H2.

### Known issues with pull request:
None

### Change log entry:
Not needed (see #9429).

### CODE REVIEW

File: source/appModules/calculator.py [hunk 1]
```diff
@@ -51,7 +55,7 @@ def event_nameChange(self, obj, nextHandler):
 		):
 			# For unit conversion, UIA notification event presents much better messages.
```
Comment (feerrenrut) [inline source/appModules/calculator.py:56]: I think this comment needs to be updated?

**Comment by josephsl** [comment]
Hi, I’m wondering the same thing. I think it would be better to remove that comment altogether, or slightly tweak it (at least for now, I think it would be better to remove it as beta cycle is approaching). Thanks.

**Comment by josephsl** [comment]
Hi, actually, I can update this comment in just a moment. Thanks.

**Review by feerrenrut — APPROVED** [review]
Thanks @josephsl

**Pull request merged**